Exclude actual token from authentication failure exception

In failures like
```
2020-07-03 19:38:43 SEVERE: Failure cause:
io.prestosql.tempto.query.QueryExecutionException: java.sql.SQLException: Authentication failed: Authentication failed for token: YIIFiQYGKwYBBQUCoIIFfTCCBXmgDTALBgkqhkiG9xIBAgKhBAMCAfaiggVgBIIFXGCCBVgGCSqGSIb3EgECAgEAboIFRzCCBUOgAwIBBaEDAgEOogcDBQAgAAAAo4IBfWGCAXkwggF1oAMCAQWhExsRTEFCUy5URVJBREFUQS5DT02iODA2oAMCAQOhLzAtGw1wcmVzdG8tc2VydmVyGxxwcmVzdG8tbWFzdGVyLmRvY2tlci5jbHVzdGVyo4IBHTCCARmgAwIBEaEDAgEBooIBCwSCAQfk1ossLMNZLs+dyxl68DNPF7W4HIKrDLVjJ9m6AFnfNCWSNguZ+LAh88zqBZYvbqs3wn13zK7N7Yn+A+tLbUxZ3LZaDIHtqdQtH1Owr9qexlBhmJymD+daDAai87uJ7LWXQnQlTVjs3ws+VNS6Lztrz51X9kdcNJGWCJucn/Q2cjLLSEi1HPBho+Bjrnow2q1GFPjlUxsSJ1KVq9sgHNxs/fYghIQ6ybofT/WreAsP80dmUAu70Veabte+9MDqUycgQPFYq/3hgOacPSLsB/Z5b42HlmVpGF9tZlay17F3oP89jvfU7cUurSn/X3VXcWLXhZL1TAko1hYQ2M20UF2a4qm1zPT+66SCA6swggOnoAMCARGiggOeBIIDmjomWBxl0ZAGleHtmggzIhOYoGLhuFBoRWoYqv5wBTTV9db3sPsfTOO+uc0CqERN76oOtoX3v3fXog1X33JWKxTzC1cDkUYOrccPSprLjC2XBDjplkyTNYR40FkAi4L/kGQhgj3kxts1tKPI8gXEr4TZFS41Vw5mF94OvWMYRZp6RzxAW7U6AREe2ZCB6ZzXJVqpxQuhnyHqDTL00Jz2W2hY3TfztiAOyzLGgHRfVgjZb8uMYV3nC4R6LsvIM4FAx2mtOEMjruUCGvBEGm99SmCVdAoXfs07P6BCejCjUX5HXzWb7x3H6Av1WnRfE6ThjqVjQhT
	at io.prestosql.tempto.query.JdbcQueryExecutor.execute(JdbcQueryExecutor.java:114)
	at io.prestosql.tempto.query.JdbcQueryExecutor.executeQuery(JdbcQueryExecutor.java:82)
	at io.prestosql.presto.tests.product.MyTest.testReads(MyTest.java:62)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.retryFailed(Invoker.java:998)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1200)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.sql.SQLException: Authentication failed: Authentication failed for token: YIIFiQYGKwYBBQUCoIIFfTCCBXmgDTALBgkqhkiG9xIBAgKhBAMCAfaiggVgBIIFXGCCBVgGCSqGSIb3EgECAgEAboIFRzCCBUOgAwIBBaEDAgEOogcDBQAgAAAAo4IBfWGCAXkwggF1oAMCAQWhExsRTEFCUy5URVJBREFUQS5DT02iODA2oAMCAQOhLzAtGw1wcmVzdG8tc2VydmVyGxxwcmVzdG8tbWFzdGVyLmRvY2tlci5jbHVzdGVyo4IBHTCCARmgAwIBEaEDAgEBooIBCwSCAQfk1ossLMNZLs+dyxl68DNPF7W4HIKrDLVjJ9m6AFnfNCWSNguZ+LAh88zqBZYvbqs3wn13zK7N7Yn+A+tLbUxZ3LZaDIHtqdQtH1Owr9qexlBhmJymD+daDAai87uJ7LWXQnQlTVjs3ws+VNS6Lztrz51X9kdcNJGWCJucn/Q2cjLLSEi1HPBho+Bjrnow2q1GFPjlUxsSJ1KVq9sgHNxs/fYghIQ6ybofT/WreAsP80dmUAu70Veabte+9MDqUycgQPFYq/3hgOacPSLsB/Z5b42HlmVpGF9tZlay17F3oP89jvfU7cUurSn/X3VXcWLXhZL1TAko1hYQ2M20UF2a4qm1zPT+66SCA6swggOnoAMCARGiggOeBIIDmjomWBxl0ZAGleHtmggzIhOYoGLhuFBoRWoYqv5wBTTV9db3sPsfTOO+uc0CqERN76oOtoX3v3fXog1X33JWKxTzC1cDkUYOrccPSprLjC2XBDjplkyTNYR40FkAi4L/kGQhgj3kxts1tKPI8gXEr4TZFS41Vw5mF94OvWMYRZp6RzxAW7U6AREe2ZCB6ZzXJVqpxQuhnyHqDTL00Jz2W2hY3TfztiAOyzLGgHRfVgjZb8uMYV3nC4R6LsvIM4FAx2mtOEMjruUCGvBEGm99SmCVdAoXfs07P6BCejCjUX5HXzWb7x3H6Av1WnRfE6ThjqVjQhT
	at io.prestosql.jdbc.PrestoStatement.internalExecute(PrestoStatement.java:271)
	at io.prestosql.jdbc.PrestoStatement.execute(PrestoStatement.java:227)
	at io.prestosql.tempto.query.JdbcQueryExecutor.executeQueryNoParams(JdbcQueryExecutor.java:122)
	at io.prestosql.tempto.query.JdbcQueryExecutor.execute(JdbcQueryExecutor.java:107)
	... 15 more
Caused by: io.prestosql.jdbc.$internal.client.ClientException: Authentication failed: Authentication failed for token: YIIFiQYGKwYBBQUCoIIFfTCCBXmgDTALBgkqhkiG9xIBAgKhBAMCAfaiggVgBIIFXGCCBVgGCSqGSIb3EgECAgEAboIFRzCCBUOgAwIBBaEDAgEOogcDBQAgAAAAo4IBfWGCAXkwggF1oAMCAQWhExsRTEFCUy5URVJBREFUQS5DT02iODA2oAMCAQOhLzAtGw1wcmVzdG8tc2VydmVyGxxwcmVzdG8tbWFzdGVyLmRvY2tlci5jbHVzdGVyo4IBHTCCARmgAwIBEaEDAgEBooIBCwSCAQfk1ossLMNZLs+dyxl68DNPF7W4HIKrDLVjJ9m6AFnfNCWSNguZ+LAh88zqBZYvbqs3wn13zK7N7Yn+A+tLbUxZ3LZaDIHtqdQtH1Owr9qexlBhmJymD+daDAai87uJ7LWXQnQlTVjs3ws+VNS6Lztrz51X9kdcNJGWCJucn/Q2cjLLSEi1HPBho+Bjrnow2q1GFPjlUxsSJ1KVq9sgHNxs/fYghIQ6ybofT/WreAsP80dmUAu70Veabte+9MDqUycgQPFYq/3hgOacPSLsB/Z5b42HlmVpGF9tZlay17F3oP89jvfU7cUurSn/X3VXcWLXhZL1TAko1hYQ2M20UF2a4qm1zPT+66SCA6swggOnoAMCARGiggOeBIIDmjomWBxl0ZAGleHtmggzIhOYoGLhuFBoRWoYqv5wBTTV9db3sPsfTOO+uc0CqERN76oOtoX3v3fXog1X33JWKxTzC1cDkUYOrccPSprLjC2XBDjplkyTNYR40FkAi4L/kGQhgj3kxts1tKPI8gXEr4TZFS41Vw5mF94OvWMYRZp6RzxAW7U6AREe2ZCB6ZzXJVqpxQuhnyHqDTL00Jz2W2hY3TfztiAOyzLGgHRfVgjZb8uMYV3nC4R6LsvIM4FAx2mtOEMjruUCGvBEGm99SmCVdAoXfs07P6BCejCjUX5HXzWb7x3H6Av1WnRfE6ThjqVjQhT
	at io.prestosql.jdbc.$internal.client.StatementClientV1.requestFailedException(StatementClientV1.java:462)
	at io.prestosql.jdbc.$internal.client.StatementClientV1.<init>(StatementClientV1.java:135)
	at io.prestosql.jdbc.$internal.client.StatementClientFactory.newStatementClient(StatementClientFactory.java:24)
	at io.prestosql.jdbc.QueryExecutor.startQuery(QueryExecutor.java:46)
	at io.prestosql.jdbc.PrestoConnection.startQuery(PrestoConnection.java:717)
	at io.prestosql.jdbc.PrestoStatement.internalExecute(PrestoStatement.java:239)
	... 18 more
```

the client may end up throwing the exception (like it happened in the test, see above), and the token value may be revealed to others as part of regular logging/reporting or feedback to the user.

we should not send the actual token back in the exception message.


See discussion @ https://prestosql.slack.com/archives/CP1MUNEUX/p1593785750079800

Starting points

https://github.com/prestosql/presto/blob/b816f299754fe966b637422c25507da2f3d3defa/presto-main/src/main/java/io/prestosql/server/security/KerberosAuthenticator.java#L155

https://github.com/prestosql/presto/blob/b816f299754fe966b637422c25507da2f3d3defa/presto-main/src/main/java/io/prestosql/server/security/KerberosAuthenticator.java#L162

--- a/presto-main/src/main/java/io/prestosql/server/security/KerberosAuthenticator.java
+++ b/presto-main/src/main/java/io/prestosql/server/security/KerberosAuthenticator.java
@@ -152,14 +152,14 @@ public class KerberosAuthenticator
                     principal = authenticate(parts[1]).orElse(null);
                 }
                 catch (RuntimeException e) {
-                    throw new RuntimeException("Authentication error for token: " + parts[1], e);
+                    throw new RuntimeException("Invalid Token", e);
                 }
             }
         }
 
         if (principal == null) {
             if (requestSpnegoToken != null) {
-                throw new AuthenticationException("Authentication failed for token: " + requestSpnegoToken, NEGOTIATE_SCHEME);
+                throw new AuthenticationException("Invalid Token", NEGOTIATE_SCHEME);
             }
             throw new AuthenticationException(null, NEGOTIATE_SCHEME);
         }

